### PR TITLE
initial prototype for modernizing js code

### PR DESCRIPTION
**Pull request by chrisguindon**
Signed-off-by: Christopher Guindon <chris.guindon@eclipse-foundation.org>

### CODE REVIEW

File: docs/index.html [hunk 1]
```diff
@@ -1183,5 +1184,17 @@ <h2 class="section-title">Other</h2>
     </footer>
     <!-- Placed at the end of the document so the pages load faster -->
     <script src="dist/js/solstice.js"></script>
+    <script>
```
Comment (chrisguindon) [inline docs/index.html:1187]: I added this to test that eclipseFdnVideos works in the global scope/window.

File: js/src/solstice.js [hunk 2]
```diff
@@ -177,7 +184,7 @@
     }, 20);
   });
 
-  eclipseFdnVideos.replace();
+  //eclipseFdnVideos();
```
Comment (chrisguindon) [inline js/src/solstice.js:187]: Same here, trying not to use this in main.js for testing

File: js/api/eclipsefdn.adopters.js [hunk 3]
```diff
@@ -134,8 +151,14 @@ class EclipseFdnAdopters {
         el[i].appendChild(h2);
       }
       const headerAnchor = document.createElement('a');
-      headerAnchor.setAttribute('class', 'btn btn-xs btn-secondary margin-left-10 uppercase');
-      headerAnchor.setAttribute('href', 'https://projects.eclipse.org/projects/' + project.project_id);
+      headerAnchor.setAttribute(
```
Comment (autumnfound) [inline js/api/eclipsefdn.adopters.js:154]: Formatter seems to be doing some weird things to js properties. We should look at how we can resolve that

**Review by ericpoirier — APPROVED** [review]

File: js/api/eclipsefdn.adopters.js [hunk 4]
```diff
@@ -168,7 +191,10 @@ class EclipseFdnAdopters {
           if (typeof adopter['logo'] !== 'undefined') {
             logo = adopter['logo'];
           }
-          if (opts['logo_white'] === true && typeof adopter['logo_white'] !== 'undefined') {
+          if (
+            opts['logo_white'] === true &&
```
Comment (autumnfound) [inline js/api/eclipsefdn.adopters.js:195]: Same as above comment, we shouldn't break Ifs up like this, just makes it take more space and sometimes be harder to read

File: js/ga/eclipsefdn.event-tracker.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      var fileExtension = fileName.split('.').pop();
+
+      // Quit here if the extension of the clicked file isn't part of the
+      // following list
+      // and if the Google Analytics is not loaded
+      var tracker = ga.getAll()[0].get('name');
+      if (
+        tracker &&
+        $.inArray(fileExtension, [
+          'pdf',
```
Comment (autumnfound) [inline js/ga/eclipsefdn.event-tracker.js:32]: While this is easier to read, I don't really see a reason for us to make this multiline. When we are looking for an extension we'd probably just do a Find for it

**Comment by chrisguindon** [comment]
@autumnfound The formatter does it. I used it to fix the inconsistent indentation.

I don't manually format code! We could look at committing a set of settings that we want for code formatting. I think that is the only scalable solution since we are more than 1 developer working on this code base.

**Comment by autumnfound** [comment]
> @autumnfound The formatter does it. I used it to fix the inconsistent indentation.
> 
> I don't manually format code! We could look at committing a set of settings that we want for code formatting. I think that is the only scalable solution since we are more than 1 developer working on this code base.

I don't either typically, its a pain! I was going to suggest we look at synchronising our settings for our formatter, so we're in agreement here!

**Comment by chrisguindon** [comment]
> > @autumnfound The formatter does it. I used it to fix the inconsistent indentation.
> > I don't manually format code! We could look at committing a set of settings that we want for code formatting. I think that is the only scalable solution since we are more than 1 developer working on this code base.
> 
> I don't either typically, its a pain! I was going to suggest we look at synchronising our settings for our formatter, so we're in agreement here!

I created a bug to discuss all of this goodness: https://github.com/EclipseFdn/solstice-assets/issues/255

**Pull request merged**